bookrack-frontend/src/components/HomePage.tsx

```diff
@@ -0,0 +1,56 @@
+import styles from "../styles/HomePage.module.css";
+import {Book} from "../assets/types/Book.ts";
+import {useEffect, useState} from "react";
+import BookCard from "./BookCard.tsx";
+import {useNavigate} from "react-router-dom";
+
+export type HomePageProps = {
+    books: Book[];
+}
+
+export default function HomePage({books}:Readonly<HomePageProps>) {
+    const [currentIndex, setCurrentIndex] = useState<number>(0)
+    const navigate = useNavigate()
+
+
+
+    useEffect(() => {
+        const timer = setInterval(() => {
+            setCurrentIndex(prevIndex => (prevIndex + 1) % books.length);
+        }, 5000)
+
+        return () => clearInterval(timer)
+
+    }, [books])
```

Comment:
What is the purpose of this timer, how does ist work?

Might be useful for my own project.

Reply:
Hi Stephan, sorry just saw this. The timer variable is used to hold the interval ID generated by the setInterval() method.

Basically, it calls the function defined within it repeatedly. 

It is used in this project to generate a slideshow of selected book images on the home page.